### PR TITLE
Fixes to the new logging configuration

### PR DESCRIPTION
**Pull request by robszewczyk**
The new logging formatting code decentralized error formatting:
different Weave sub-libraries -- WeaveSystem, Inet, Warm -- now
provide their own error formatting.  Within Weave, however, it is
desirable to have a single config flag that switches between short and
verbose error codes.  To that end, we retain the control over the
logging format via WEAVE_CONFIG_SHORT_ERR_STRING, and to ensure the
propagation of that define into appropriate modules, we include the
config headers that could impact the value of that define in the
SystemConfig.h.

Include <stddef.h> in the new error formatter files, to ensure that
NULL is defined.

### CODE REVIEW

File: src/system/SystemConfig.h [hunk 1]
```diff
@@ -40,6 +40,33 @@
 /* Platform include headers */
 #include <BuildConfig.h>
 
+/* Include a project-specific configuration file, if defined.
```
Comment (jaylogue) [inline src/system/SystemConfig.h]: Maybe we should differentiate this comment from the similar one below.  E.g. :
  
_Include a project-specific **Weave** configuration file file, if defined._

File: src/system/SystemConfig.h [hunk 2]
```diff
@@ -40,6 +40,33 @@
...
+ * application-specific values.  The WeaveProjectConfig.h file is typically
+ * located outside the OpenWeave source tree, alongside the source code for the
+ * application.  The config file is included here to enable certain system-wide
+ * configuration options, primarily related to logging and error reporting.
+ */
+#ifdef WEAVE_PROJECT_CONFIG_INCLUDE
+#include WEAVE_PROJECT_CONFIG_INCLUDE
+#endif
+
+/* Include a platform-specific configuration file, if defined.
```
Comment (jaylogue) [inline src/system/SystemConfig.h]: Some comment as above.

File: src/system/SystemConfig.h [hunk 3]
```diff
@@ -40,6 +40,33 @@
...
+ * adapts OpenWeave to the target environment.  This adaptation layer may be
+ * included in the OpenWeave source tree itself or implemented externally.  The
+ * config file is included here to enable certain system-wide configuration
+ * options, primarily related to logging and error reporting.
+ */
+#ifdef WEAVE_PLATFORM_CONFIG_INCLUDE
+#include WEAVE_PLATFORM_CONFIG_INCLUDE
+#endif
+
 /* Include a project-specific configuration file, if defined.
```
Comment (jaylogue) [inline src/system/SystemConfig.h]: Some comment as above.

File: src/system/SystemConfig.h (hunk 1 above)
Comment (robszewczyk) [reply src/system/SystemConfig.h]: done

File: src/system/SystemConfig.h (hunk 2 above)
Comment (robszewczyk) [reply src/system/SystemConfig.h]: done

File: src/system/SystemConfig.h (hunk 3 above)
Comment (robszewczyk) [reply src/system/SystemConfig.h]: done

**Pull request merged**